modules/seqexec/engine/src/test/scala/seqexec/engine/SequenceSpec.scala

```diff
@@ -146,25 +146,36 @@ class SequenceSpec extends FlatSpec {
   private val result: Result[Nothing] = Result.OK(DummyResult)
   private val action: Action[IO] = fromF[IO](ActionType.Undefined, IO(result))
   private val completedAction: Action[IO] = action.copy(state = Action.State(Action.ActionState.Completed(DummyResult), Nil))
-  def simpleStep2(pending: List[Actions[IO]], focus: Execution[IO], done: List[Results[IO]]): Step.Zipper[IO] = {
-    val rollback: (Execution[IO], List[Actions[IO]]) =  done.map(_.map(const(action))) ++ List(focus.execution.map(const(action))) ++ pending match {
-      case Nil => (Execution.empty, Nil)
-      case x::xs => (Execution(x), xs)
+  def simpleStep2(pending: List[ParallelActions[IO]], focus: Execution[IO], done: List[NonEmptyList[Result[IO]]]): Step.Zipper[IO] = {
+    val rollback: (Execution[IO], List[ParallelActions[IO]]) =  {
+      val doneParallelActions: List[ParallelActions[IO]] = done.map(_.map(const(action)))
+      val focusParallelActions: List[ParallelActions[IO]] = focus.toParallelActionsList
+      doneParallelActions ++ focusParallelActions ++ pending match {
+        case Nil => (Execution.empty, Nil)
+        case x::xs => (Execution(x.toList), xs)
+      }
     }
 
-    Step.Zipper(1, breakpoint = Step.BreakpointMark(false), Step.SkipMark(false), pending, focus, done.map(_.map{r =>
-      val x = fromF[IO](ActionType.Observe, IO(r))
-      x.copy(state = Execution.actionStateFromResult(r)(x.state))
-    }), rollback)
+    Step.Zipper(
+      id = 1,
+      breakpoint = Step.BreakpointMark(false),
+      Step.SkipMark(false),
+      pending = pending,
+      focus = focus,
+      done = done.map(_.map{r =>
+        val x = fromF[IO](ActionType.Observe, IO(r))
+        x.copy(state = Execution.actionStateFromResult(r)(x.state))
+      }),
+      rollback)
```

Comment:
Just being pedantic and stating that you missed `rolledback = ` here. 😄

Reply:
👍